bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/storage/ldb/SingleDirectoryDbLedgerStorage.java

```diff
@@ -155,6 +157,9 @@ public SingleDirectoryDbLedgerStorage(ServerConfiguration conf, LedgerManager le
         readCacheMaxSize = readCacheSize;
         readAheadCacheBatchSize = conf.getInt(READ_AHEAD_CACHE_BATCH_SIZE, DEFAULT_READ_AHEAD_CACHE_BATCH_SIZE);
 
+        // Do not attempt to perform read-ahead more than half the total size of the cache
+        maxReadAheadBytesSize = readCacheMaxSize / 2;
```

Comment:
Do we need to consider the multiple ledger directories?

Reply:
It's already being accounted in that `readCacheMaxSize` size is divided evenly divided by the directories.

Reply:
ok